cdap-distributions/src/hdinsight/pageblob-configure.sh

```diff
@@ -0,0 +1,142 @@
+#!/bin/bash
+#
+# Copyright © 2016 Cask Data, Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License"); you may not
+# use this file except in compliance with the License. You may obtain a copy of
+# the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+# WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+# License for the specific language governing permissions and limitations under
+# the License.
+
+#
+# Configure HDFS Pageblob support for CDAP HDInsight cluster
+#
+
+# Ambari constants
+AMBARICONFIGS_SH=/var/lib/ambari-server/resources/scripts/configs.sh
+AMBARIPORT=8080
+ACTIVEAMBARIHOST=headnodehost
+
+# Function definitions
+
+die() { echo "ERROR: ${*}"; exit 1; };
+
+checkHostNameAndSetClusterName() {
+    fullHostName=$(hostname -f)
+    echo "fullHostName=$fullHostName"
+    CLUSTERNAME=$(sed -n -e 's/.*\.\(.*\)-ssh.*/\1/p' <<< $fullHostName)
+    if [ -z "$CLUSTERNAME" ]; then
+        CLUSTERNAME=$(echo -e "import hdinsight_common.ClusterManifestParser as ClusterManifestParser\nprint ClusterManifestParser.parse_local_manifest().deployment.cluster_name" | python)
+        if [ $? -ne 0 ]; then
+            echo "[ERROR] Cannot determine cluster name. Exiting!"
+            exit 133
+        fi
+    fi
+    echo "Cluster Name=$CLUSTERNAME"
+}
+
+# Try to perform an Ambari API GET
+# Dies if credentials are bad
+__validateAmbariConnectivity() {
+    coreSiteContent=$(bash ${AMBARICONFIGS_SH} -u ${USERID} -p ${PASSWD} get ${ACTIVEAMBARIHOST} ${CLUSTERNAME} core-site)
+    if [[ ${coreSiteContent} == *"[ERROR]"* && ${coreSiteContent} == *"Bad credentials"* ]]; then
+        echo '[ERROR] Username and password are invalid. Exiting!'
+        exit 134
+    fi
```

Comment:
- No need for using `bash` in subshell, since scripts are executable and have shebangs
- I prefer `=~` and using regex syntax, but I'll leave that to you
- Use new die